game-core/src/main/java/games/strategy/engine/random/MartiDiceRoller.java

```diff
@@ -96,16 +95,15 @@ public String postRequest(
   }
 
   @Override
-  public int[] getDice(final String string, final int count)
-      throws IOException, InvocationTargetException {
+  public int[] getDice(final String string, final int count) throws DiceServerException {
     final Matcher errorMatcher = errorPattern.matcher(string);
     if (errorMatcher.find()) {
-      throw new InvocationTargetException(null, errorMatcher.group(1));
+      throw new DiceServerException(errorMatcher.group(1));
     }
 
     final Matcher diceMatcher = dicePattern.matcher(string);
     if (!diceMatcher.find()) {
-      throw new IOException("String '" + string + "' has an invalid format.");
+      throw new IllegalStateException("String '" + string + "' has an invalid format.");
```

Comment:
Nit, we should keep in mind the audience of exception messages. Things like "String" is geared towards a developer, would not otherwise make sense to an end user. They want to know what happened and what they should do about it if anything.

With that said, this exception type and message are fine for now. We may though want to consider creating some infrastructure that would suffix an error message with something like "This is an unexpected error and indicates a bug. Please report this to TripleA development"